CMakeLists.txt

```diff
@@ -428,19 +424,16 @@ endif()
 # Do tests after chrpath so that we use the "real" cuda driver
 add_subdirectory(tests)
 
-# AUTO_INSTALL_DIR -> Optional: specify post-build install direcory
-if(AUTO_INSTALL_DIR)
```

Comment:
Please leave this in. This causes deployment without calling the 'install' target, which is desired in some cases.  If AUTO_INSTALL_DIR is not defined, it will not execute.
If this was removed in dmlc-core, please return that as well.

Reply:
I don't understand. If you don't want to call the install target, why not simply not run `make install`? Running `make` or `make all` will not run the install target anyway.

Reply:
There are several development workflows where running make install is not desired and header outputs need to be distributed automatically during a normal build.  One of them is the Amazon Brazil build system. Another is when building with CLion and the submodules are not positioned heiarchally under mxnet. No elegant way was found to trigger "install" with a normal build. I don't think they interfere with your normal "install" code that you added.

Reply:
True, there is no conflict. I originally removed these lines because I figured I replaced them with cmake `install()` commands.

I do think cmake `install(... COMPONENT <name>)` together with customizing installation prefixes should be flexible enough for practically any situation. And it would have the advantage of having only one set of install rules.

Either way, they're back now.